### PR TITLE
update formatting of variant allele and variant locations

### PR DESCRIPTION
**Pull request by jyothishnt**
## Description
1. Replace / with space for allele string
2. Remove end location from variant location

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2186

## Deployment URL(s)
http://format-allele.review.ensembl.org
http://format-allele.review.ensembl.org/genome-browser/grch38?focus=variant:1:230710047:rs1553314015&location=1:230710032-230710064
Long allele: http://format-allele.review.ensembl.org/genome-browser/grch38?focus=variant:1:244874375:rs34828567&location=1:244874345-244874398

## Views affected
GB - Variant

### CODE REVIEW

**Comment by azangru** [comment]
I don't know if we want this comma. When the alleles are written in a single line, it doesn't show 🤷‍♂️ 

image

File: src/shared/components/variant-alleles-sequences/VariantAllelesSequences.scss [hunk 1]
```diff
@@ -1,3 +1,7 @@
 .expanded span {
   display: block;
+}
+
+.alleleSequence {
```
Comment (azangru) [inline src/shared/components/variant-alleles-sequences/VariantAllelesSequences.scss:5]: This whole component is about allele sequences 🙂 Let's try to think what the element with this class means. Is it, perhaps, the compact one-line representation of the allele sequences? `.compact`?

File: src/shared/components/variant-alleles-sequences/VariantAllelesSequences.tsx [hunk 2]
```diff
@@ -108,7 +108,9 @@ const prepareSequenceData = (props: Props) => {
     alleleSequences.unshift(referenceAllele.reference_sequence);
   }
 
-  const combinedString = alleleSequences.join('/');
+  const alleleSequencesCopy = alleleSequences.slice(0);
+  const combinedString =
+    alleleSequencesCopy.shift() + '  ' + alleleSequencesCopy.join(',');
```
Comment (azangru) [inline src/shared/components/variant-alleles-sequences/VariantAllelesSequences.tsx:113]: This line shows that you are trying to separate the reference allele from the alternative alleles. But we already had this separation above, and lost it on line 108. The fact that we need to keep the reference allele and alternative alleles separately suggests that it is worth rewriting the function, like so:

```
const prepareSequenceData = (props: Props) => {
  const { referenceAllele, alternativeAlleles } = getReferenceAndAltAlleles(
    props.alleles
  );

  // we expect reference allele to always exist in the data;
  // but default to an empty string to appease typescript
  const referenceAlleleSequence = referenceAllele?.reference_sequence ?? '';

  const alternativeAlleleSequences = alternativeAlleles.map(
    (allele) => allele.allele_sequence
  );

  const combinedString = `${referenceAlleleSequence} ${alternativeAlleleSequences.join(',')}`;

  return {
    referenceAlleleSequence,
    alternativeAlleleSequences,
    combinedString
  };
};
```

And then you can also update the `ExpandedView` component to treat the `referenceAlleleSequence` and the `alternativeAlleleSequences` differently. This makes the programmer's intention clearer than just treating the zeroth element of a string array differently.

File: src/shared/components/variant-alleles-sequences/VariantAllelesSequences.scss (hunk 1 above)
Comment (jyothishnt) [reply src/shared/components/variant-alleles-sequences/VariantAllelesSequences.scss:5]: or `.trimmedSequence` ?

File: src/shared/components/variant-alleles-sequences/VariantAllelesSequences.tsx [hunk 3]
```diff
@@ -98,20 +109,21 @@ const prepareSequenceData = (props: Props) => {
     props.alleles
   );
 
-  const alleleSequences = alternativeAlleles.map(
+  // we expect reference allele to always exist in the data;
+  // but default to an empty string to appease typescript
```
[the remark this answers is not in the document]
Comment (azangru) [reply src/shared/components/variant-alleles-sequences/VariantAllelesSequences.tsx:113]: Notice the `?? ''` below 🙂

[the remark this answers is not in the document]
Comment (imransl) [reply src/shared/components/variant-alleles-sequences/VariantAllelesSequences.tsx:113]: That is not too explicit as the `if` check though? What I am wondering is whether this comment is a must-to-have here anymore.

[the remark this answers is not in the document]
Comment (azangru) [reply src/shared/components/variant-alleles-sequences/VariantAllelesSequences.tsx:113]: This update to the function comes from the [comment above](https://github.com/Ensembl/ensembl-client/pull/1029#discussion_r1351989762), where I tried to explain why we are doing the `?? ''`. On the one hand, reference allele should always be present in the list of variant alleles. On the other hand, what if something happens and it is not?

[the remark this answers is not in the document]
Comment (jyothishnt) [reply src/shared/components/variant-alleles-sequences/VariantAllelesSequences.tsx:113]: Is this still an issue @imransl ?

[the remark this answers is not in the document]
Comment (imransl) [reply src/shared/components/variant-alleles-sequences/VariantAllelesSequences.tsx:113]: No it isn't

**Review by imransl — APPROVED** [review]

**Review by azangru — APPROVED** [review]

**Pull request merged**